pages/api/course/index.ts

```diff
@@ -0,0 +1,58 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import prisma from "../../../lib/prisma";
+
+export default async function handle(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method === "POST") {
+    if (!req.body.name) {
+      return res.status(400).json({ error: "Missing course name" });
+    }
+
+    if (!req.body.lat || !req.body.lon || !req.body.city || !req.body.state) {
+      return res.status(400).json({ error: "Missing course location" });
+    }
+
+    const name = req.body.name as string;
+    const lat = req.body.lat as number;
+    const lon = req.body.lon as number;
+    const city = req.body.lat as string;
+    const state = req.body.lat as string;
+
+    const sameLocation = await prisma.location.findFirst({
+      where: { city: city, state: state },
+    });
+    if (sameLocation) {
+      const courseWithName = await prisma.course.findFirst({
+        where: { name: name, location: sameLocation },
+      });
+      if (courseWithName) {
+        return res
+          .status(409)
+          .json({ error: "Course with same name in same location exists" });
+      }
+    }
+
+    const location = await prisma.location.create({
```

Comment:
Are we expecting the user to manually type in `city` and `state`? Might need some toLowerCase if that is the case.

Reply:
I wasn't sure the best way to go about this, I was thinking either they could enter it in or we could try to find an API that allows a lookup or even go straight to a map API that would allow a user to drop a pin that was give coordinates, city, and state. I didn't want to commit to doing it in the backend but if we got the coordinates we probably could

Reply:
I agree with lower case long them tho

Reply:
Yeah, when I'll probably try to find some free (non-Google maps probably) map input that will be able to give us the `lat` and `lng`. Hopefully it will also give the city and state, but I'll circle back to this after I add that to a course creation page.